my-app/src/components/atoms/input-text/input-text.tsx

```diff
@@ -3,44 +3,59 @@ import { clsx } from 'clsx';
 
 interface InputTextProps {
     className?: string;
-    defaultInputValue?: string;
+    formikValue?: string | number;
     placeholder?: string;
     label?: string;
-    onlyNumber?: boolean
+    onlyNumber?: boolean;
+    formikHandlerChange?: (e: React.ChangeEvent<HTMLInputElement>, r: boolean ) => void;
+    name?: string;
+    formikError?: any;
 }
 
 export const InputText = ({
     className,
-    defaultInputValue,
+    formikValue,
     placeholder,
     label,
-    onlyNumber
+    onlyNumber,
+    formikHandlerChange,
+    name,
+    formikError
 }: InputTextProps) => {
     const [inputValue, setInputValue] = useState('');
-    const handlerChange = (e: React.ChangeEvent<HTMLInputElement>) => setInputValue(e.target.value);
-    const handlerChangeNumber = (e: React.ChangeEvent<HTMLInputElement>) => {
-        const re = new RegExp('^[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)$');
-        if (e.target.value === '' || re.test(e.target.value)) {
+    const handlerChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (onlyNumber) {
+            const re = new RegExp('^[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)$');    
```

Comment:
There is duplicated functionality for numbers only here and in add-movie-form, I'd propose to rewrote it here to 


  const handlerChange = (e: React.ChangeEvent<HTMLInputElement>) => {

        const setValue = formikHandlerChange ? formikHandlerChange(e, onlyNumber) : setInputValue;
        
        if (onlyNumber) {
            const re = new RegExp('^[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)$');    

            if (e.target.value === '' || re.test(e.target.value)) {
                setValue(e.target.value);
            }
        } else {
            setValue(e.target.value)
        }
    };

Reply:
And in add-movie-form:


const formikHandlerChange = (e: React.ChangeEvent<HTMLInputElement>, onlyNumber: boolean) => {
        
const inputId = e.currentTarget.id;

        return (value: string) => {
            formik.setFieldValue(inputId, onlyNumber ? parseInt(value) : value);
        }
    }`

Reply:
Done